synnax/pkg/distribution/channel/lease_proxy.go

```diff
@@ -137,17 +142,65 @@ func (lp *leaseProxy) createFreeVirtual(
 	if lp.freeCounter == nil {
 		panic("[leaseProxy] - tried to assign virtual keys on non-bootstrapper")
 	}
-	toCreate, err := lp.maybeRetrieveExisting(ctx, tx, channels, lp.freeCounter, retrieveIfNameExists)
+
+	// If existing channels are passed in, update as necessary (for calc channels)
+	// unless retrieveIfNameExists is true,
+	//in which case we just update returned list of channels
+	if retrieveIfNameExists {
+		var found []Channel
+		err := gorp.NewRetrieve[Key, Channel]().
+			WhereKeys(KeysFromChannels(*channels)...).
+			Entries(&found).
+			Exec(ctx, tx)
+
+		if err != nil && !errors.Is(err, query.NotFound) {
+			return err
+		}
+		for _, f := range found {
+			for i, ch := range *channels {
+				if ch.Key() == f.Key() {
```

Comment:
Have you taken a look at `lo.Includes`? There is pretty deep nesting here, and it could be useful to extract the `KeysFromChannels` call on 152 into a variable called `keys` and then you do lo.Includes(keys, f.Key() to save yourself a layer of nesting.

Reply:
Isn't this block of code redundant? Doesn't `retrieveAndAssignExistingKeys` already do this?

Reply:
it doesn't. at least when i remove this code, the test I have to check this fails

Reply:
Let's take a look at this together. I think there's a way to condense this code.